libs/libqmi/Makefile

```diff
@@ -59,26 +64,31 @@ define Package/libqmi-utils/description
   Utils to talk to QMI enabled modems
 endef
 
-CONFIGURE_ARGS += \
-	--disable-static \
-	--disable-gtk-doc \
-	--disable-gtk-doc-html \
-	--disable-gtk-doc-pdf \
-	--disable-silent-rules \
-	--enable-firmware-update \
-	--without-udev \
-	--without-udev-base-dir
+MESON_ARGS += \
+	-Dudev=false \
+	-Dintrospection=false \
+	-Dman=false \
+	-Dbash_completion=false \
+	-Db_lto=true
 
 ifeq ($(CONFIG_LIBQMI_WITH_MBIM_QMUX),y)
-	CONFIGURE_ARGS += --enable-mbim-qmux
+	MESON_ARGS += -Dmbim_qmux=true
 else
-	CONFIGURE_ARGS += --disable-mbim-qmux
+	MESON_ARGS += -Dmbim_qmux=false
 endif
 
 ifeq ($(CONFIG_LIBQMI_WITH_QRTR_GLIB),y)
-	CONFIGURE_ARGS += --enable-qrtr
+	MESON_ARGS += -Dqrtr=true
+else
+	MESON_ARGS += -Dqrtr=false
+endif
+
+ifeq ($(CONFIG_LIBQMI_COLLECTION_MINIMAL),y)
```

Comment:
I'm not sure about having separate config fields for these, as they're mutually exclusive.
Can we not have a single `CONFIG_LIBQMI_COLLECTION` field and then the user can select between "minimal", "basic" and "full"?

Reply:
> No, default should be FULL, or otherwise we'll get tons of complaints.

Do you mean the reduced functionality of the library? But as I know basic is enougth for modemmanager. There is no any packages which also uses libqmi. OpenWrt works on different devices. Most of them have a limited amount of resources. This makes things a little better.

Reply:
> But as I know basic is enough for modemmanager.

Yes, "basic" is what MM expects. But there may be others integrating other custom applications out there that rely on qmicli commands not included in the basic collection. I'm not fully convinced that we should use that as default. @nickberry17 what's your take on this?

Reply:
> I didn't find better solutions. It would be great if someone could point me a better way.

Switching to a single "string" type config option could be a way to do that, with "full" as default. And then explain that libqmi provides 3 different collections by default, "minimal", "basic" and "full".

The name of the collection is not limited to those 3 in libqmi, because users could also use a custom different collection e.g. named "something-else", as long as `data/qmi-collection-something-else.json` file exists in the tree.

Reply:
> Switching to a single "string" type config option could be a way to do that

But there is no choice in this case. The user will select a collection by typing its name manually. This may result in an typo error. There are three collections. It is quite logical to have a choice between them. It's not that hard to add a 4th choice, where the user himself can specify the collection manually. I'd also like to hear @nickberry17 opinion.

Reply:
This can be simplified to a single line.

Reply:
So now we have 3 different boolean options that are mutually exclusive. Is that really a thing? Shouldn't we at least try to e.g. error out somewhere if the user sets 2 booleans to TRUE or something?

Reply:
"minimal" is contained in "basic". "basic" is contained in "full".

Selecting this config wouldn't make any sense, and it would completely ignore the FULL request and only enable the MINIMAL collection:
```
CONFIG_LIBQMI_COLLECTION_MINIMAL=y
CONFIG_LIBQMI_COLLECTION_BASIC=n
CONFIG_LIBQMI_COLLECTION_FULL=y
```
There has to be a better way to handle this... or maybe there isn't?

Reply:
@aleksander0m is there any harm in passing all of these at build time?  Because if that is okay, then one thing we could do is just make each selection progressively select all the other ones that are a subset of it using `select`.  I haven't tested this but it looks like it may work.

For example:
https://github.com/openwrt/packages/blob/38f01ad2c9738c6c5a5dbda85019bb9cf5bc10f8/utils/lcd4linux/Config.in#L220

```
config LCD4LINUX_CUSTOM_DRIVER_picoLCDGraphic
	bool
	prompt "picoLCDGraphic"
	select LCD4LINUX_CUSTOM_NEEDS_libgd
	select LCD4LINUX_CUSTOM_NEEDS_libusb
```

@neheb  what do you think?

Reply:
auto-selecting CONFIG_LIBQMI_COLLECTION_MINIMAL if CONFIG_LIBQMI_COLLECTION_BASIC is selected, and auto-selecting CONFIG_LIBQMI_COLLECTION_BASIC if CONFIG_LIBQMI_COLLECTION_FULL is selected, would make a lot of sense config-wise.

Then, the Makefile could look for those options in reverse order, first check for FULL, then check for BASIC, and then check for MINIMAL. Right now it's looking at those options in the reverse order.

Reply:
@Linaro1985 what do you think of that suggestion?